src/encoded/static/components/genome_browser.js

```diff
@@ -118,30 +120,35 @@ class GenomeBrowser extends React.Component {
     }
 
     componentDidMount() {
-        // Determine pinned files based on genome, filter and sort files, compute and draw tracks
-        this.setGenomeAndTracks();
+        // Check if browser is IE 11 and disable browser if so
+        if (BrowserFeat.getBrowserCaps('uaTrident')) {
+            this.setState({ disableBrowserForIE: true });
+        } else {
+            // Determine pinned files based on genome, filter and sort files, compute and draw tracks
+            this.setGenomeAndTracks();
+        }
     }
 
     componentDidUpdate(prevProps, prevState) {
-        if (this.state.contig !== prevState.contig) {
+        if (this.state.contig !== prevState.contig && !(this.state.disableBrowserForIE)) {
```

Comment:
Would it be better to have just one check for !(this.state.disableBrowserForIE) in an `if` that wrapped around all this, or the opposite check that returns immediately? It seems like you skip everything in this method if this.state.disableBrowserForIE is true anyway. Also, I do wonder if componentDidUpdate would _ever_ get called if componentDidMount doesn’t call setGenomeAndTracks(), and so you wouldn’t need this check at all? I get it if you’d like to check anyway, for safety.

Reply:
Oh yes, that seems easier. I remember thinking about that and then adding it to every conditional instead, but I think this way there will only be one check if the browser is IE and that's better. Also, this check is definitely necessary. The component updates (1) when you change the screen size, say by clicking on the arrow that expands or collapses the facets, (2) every time you select a facet or assembly, and (3) when you click from tab to tab. So everything will break if we don't keep this!